File: src/Java/algorithms/arrandstring/FizzBuzz.java
package algorithms.arrandstring; 

import java.util.*; 

public class FizzBuzz {
    /* Function: When reach the multiple of 3, say fizz; when reach the multiple of 5, say buzz. 
     * Solution: If case. 
     */ 
    
    public String fizzbuzz(int n) {
        if (n <= 0) {
            return ""; 
        }
        
        StringBuffer sb = new StringBuffer(); 
        for (int i = 1; i <= n; ++i) {
            if (i % 3 == 0) {
                sb.append("Fizz"); 
            }
            if (i % 5 == 0) {
                sb.append("Buzz"); 
            }
            if (i % 3 != 0 && i % 5 != 0) {
                sb.append(i); 
            }
            
            if (i != n) {
                sb.append(", "); 
            }
        }
        
        return sb.toString(); 
    }
} 

File: src/CPP/src/leetcode/FizzBuzz.cpp
/*
 * [Source] https://leetcode.com/problems/fizz-buzz/
 * [Difficulty]: Easy
 * [Tag]: 
 */

#include <iostream>
#include <vector>
#include <string>

using namespace std;

// [Solution]:
// [Corner Case]:
class Solution {
public:
    vector<string> fizzBuzz(int n) {
        if (n <= 0)
            return {};

        vector<string> res;
        for (int i = 1; i <= n; ++i) {
            string tmp;
            if (i % 3 == 0)
                tmp += "Fizz";
            if (i % 5 == 0)
                tmp += "Buzz";
            if (tmp == "")
                tmp = to_string(i);
            res.push_back(tmp);
        }
        return res;
    }
};

// [Solution]:
/* Java solution

 */

int main() {
    Solution sol;

    for (string &s : sol.fizzBuzz(15))
        cout << s << " ";
    cout << endl;

    return 0;
}
